File: e808.js
const {primes, reverse, isprime} = require('./myfunc');

const P = primes(5000000);
let s = k = 0;
for (let i = 0; i < P.length; i++) {
    const n = P[i]**2;
    const m = reverse(n);
    if (n !== m) {
        const r = Math.sqrt(m);
        if (r === Math.floor(r) && isprime(r)) {
            k++;
            s += n;
        }
        if (k === 50) break;
    }
}
console.log(s);

File: e808.py
# Problem 808 - Reversible prime squares

from myfunc import isprime, primes
from math import sqrt
from time import time

tic = time()
P = primes(31100101)
print(f'Primes table generated in {time() - tic} seconds')
s = 0
k = 0
for i in P:
    n = i*i
    m = int(str(n)[::-1])
    if (n != m):
        r = sqrt(m)
        if r == int(r) and isprime(r):
            k += 1
            s += n
            print(k, n, int(sqrt(n)))
        if k == 50:
            break
print(s)

print(f'Run time: {time() - tic} seconds')
